src/dbnode/storage/shard.go

```diff
@@ -1283,7 +1290,10 @@ func (s *dbShard) insertSeriesBatch(inserts []dbShardInsert) error {
 
 		if inserts[i].opts.hasPendingWrite {
 			write := inserts[i].opts.pendingWrite
-			err := entry.Series.Write(ctx, write.timestamp, write.value,
+			// NB: Since this is adding a new series, it will always write a new value
```

Comment:
I would change this comment to say: "We ignore the `wasWritten` return argument because this is an async operation and there is nothing to do with the value right now. In the future, we could try and propagate the `wasWritten` argument back to the caller somehow using the waitgroup, but its not required right now."